source/marsGameSound.js

```diff
@@ -20,15 +20,15 @@ this.initialize = function() {
     for ( var i = 0; i < soundSet.length; ++i ) {
         this.loadSound( soundSet[i] );
     }
-    this.future( 0 ).setUpSubtitles();
+    this.future( 0.01 ).setUpSubtitles();
```

Comment:
Okay, I think you would find that if you replaced every `future( 0 )` call with a `future( 0.01 )` call, you would be right back in the same situation you were before.  Basically, `future( 0 )` calls are placed on the queue in the order they are "scheduled" (when that line of code executes).  Since when the application is created, `initialize` calls are executed from the bottom up, children `future( 0 )` calls will all get scheduled before parent `future( 0 )` calls.  If a parent `future( 0 ) call needs to run first, then the child call should be scheduled for a later time (`future( 0.01)`).  This could get really complicated with a series of `future( 0 )`, `future( 0.01)`, `future( 0.02 )`, etc calls, but in practice we haven't run into that complicated of dependency structures.  If that ended up happening, there is probably a better way to do what needs to be done that should be investigated.